workers/data_refinery_workers/processors/utils.py

```diff
@@ -668,3 +670,48 @@ def handle_processor_exception(job_context, processor_key, ex):
     job_context["job"].failure_reason = err_str
     job_context["success"] = False
     return job_context
+
+
+def cache_keys(*keys, work_dir_key='work_dir'):
+    """ Decorator to be applied to a pipeline function.
+    Returns a new function that calls the original one and caches the given
+    keys into the `work_dir`. On the next call it will load those keys (if they
+    exist) and add them to the job_context instead of executing the function. """
+    def inner(func):
+        key_names = ''.join(list(keys) + [func.__name__])
```

Comment:
Hmm, just for readability if we ever want to inspect these caches or something.... 
```suggestion
        key_names = '__'.join(list(keys) + [func.__name__])
```

Reply:
hmm I wonder if the filename of the cache should be this, instead of the `md5` hash, so that they are readable. This might be better :)

Reply:
Oh yeah I missed that you were doing a md5 hash, yeah I like this!